crates/invoker-impl/src/invocation_task.rs

```diff
@@ -409,46 +405,146 @@ where
             shortcircuit!(tokio::try_join!(read_journal_future, read_state_future));
 
         // Resolve the deployment metadata
-        let (deployment, deployment_changed) =
-            if let Some(deployment_id) = journal_metadata.deployment_id {
+        let (deployment, chosen_service_protocol_version, deployment_changed) =
+            if let Some(chosen_deployment) = &journal_metadata.chosen_deployment {
                 // We have a pinned deployment that we can't change even if newer
                 // deployments have been registered for the same service.
                 let deployment_metadata = shortcircuit!(self
                     .deployment_metadata_resolver
-                    .get_deployment(&deployment_id)
-                    .ok_or_else(|| InvocationTaskError::UnknownDeployment(deployment_id)));
-                (deployment_metadata, /* has_changed= */ false)
+                    .get_deployment(&chosen_deployment.deployment_id)
+                    .ok_or_else(|| InvocationTaskError::UnknownDeployment(
+                        chosen_deployment.deployment_id
+                    )));
+
+                // todo: We should support resuming an invocation with a newer protocol version if
+                //  the endpoint supports it
```

Comment:
Not sure if this would be responsability of the PP or of the invoker to be able to transcode the journal from an old to a new version.

Reply:
I think there are two aspects to the answer: The wire format (invoker to service endpoint) should be independent of how we store the records in the PP. Given this, I think it is the responsibility of the invoker to create the correctly encoded messages for the service endpoint based on the chose service protocol.